### ISSUE TITLE
Refactor existing toString(), equals() and hashCode() methods to consistently use the Apache Commons Lang 3 style

### ISSUE THREAD

**Issue by t616178** [gchq/Gaffer#864]
All classes which implement toString(), equals() and hashCode() in the Gaffer codebase should do in a consistent way. We have chosen the Apache Commons Lang style for use throughout the project.

Classes which implement these methods should contain implementations similar to that described below:

```java
    @Override
    public boolean equals(final Object obj) {
        if (this == obj) {
            return true;
        }

        if (obj == null || getClass() != obj.getClass()) {
            return false;
        }

        final MyObject myObj = (MyObject) obj;

        return new EqualsBuilder()
                .append(field1, myObj.field1)
                .append(field2, myObj.field2)
                .isEquals();
    }

    @Override
    public int hashCode() {
        return new HashCodeBuilder(17, 37)
                .append(field1)
                .append(field2)
                .toHashCode();
    }

    @Override
    public String toString() {
        return new ToStringBuilder(this)
                .append("field1", field1)
                .append("field2", field2)
                .toString();
    }
```

**Comment by GCHQDeveloper404** [comment]
toString portion of this ticket has been done.

**Comment by goti921** [comment]
@t616178 
Is it up for grabs ?

**Comment by p013570** [comment]
Hi @goti921,

Yeah go for it, thanks! 

@GCHQDeveloper404 has completed the toString part of the issue, I'm not sure how much is actually left to do. Please branch off the ```develop``` branch and refer to our [ways of working](https://github.com/gchq/Gaffer/wiki/Ways-of-Working#pull-requests) wiki page.

Let me know if you have any questions.

**Comment by m55624** [comment]
I am going to start work on this.  @goti921 please let me know if you have done anything on this and where and I will stop.

**Comment by goti921** [comment]
Hi @m55624 
I have almost completed refactoring of the equals(0 method. 
You can carry on with the hashcode() method.

**Comment by m55624** [comment]
@goti921 
I will leave this for now.  Thanks for doing this!

**Comment by p013570** [comment]
Merged into develop-1.0.0.
